### PR TITLE
envvars becomes var, can also store vars for connections

### PR DESCRIPTION
**Pull request by itayd**

### CODE REVIEW

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [inline internal/backend/db/dbgorm/vars.go:21]: please split all relevant functions to SDK/schema for testability and add `vars_test.go` (see env_tests.go removed lines)

**Comment by ig-ra** [comment]
I think we should try to open much smaller PRs, i.e. ~300 lines (excluding auto-generated files, of coz).
For example in this PR all db-related changes (EnvVar -> Var) could be extracted and handled in separate PR.

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [inline internal/backend/db/dbgorm/vars.go]: ns -> names?

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [inline internal/backend/db/dbgorm/vars.go:25]: let's add a string to make this error more human-friendly?
fmt.Errorf("setVars: #%d: %w", i, err)

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [inline internal/backend/db/dbgorm/vars.go:36]: could we enclose in db.transaction only db-related ops?
e.g. first create (not under transaction) list of scheme.Var with everything precomputed and separated list of indexes + cids to fetch iid(s), 
then under transcation only fetch and update?

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [inline internal/backend/db/dbgorm/vars.go]: should we check sid.IsValid() and return error? this is what we did in EnvVars

File: cmd/ak/cmd/vars/delete.go
Comment (ig-ra) [inline cmd/ak/cmd/vars/delete.go]: delete?

File: internal/backend/db/dbgorm/vars.go
Comment (itayd) [reply internal/backend/db/dbgorm/vars.go:21]: https://linear.app/autokitteh/issue/ENG-768/db-vars-testing

File: internal/backend/db/dbgorm/vars.go
Comment (itayd) [reply internal/backend/db/dbgorm/vars.go]: done

File: internal/backend/db/dbgorm/vars.go
Comment (itayd) [reply internal/backend/db/dbgorm/vars.go:25]: i think that adding the `setVars` perfix is wrong, the calling function should add it.

File: cmd/ak/cmd/vars/delete.go
Comment (itayd) [reply cmd/ak/cmd/vars/delete.go]: ?

Comment (itayd) [reply cmd/ak/cmd/vars/delete.go]: oh right! done

File: internal/backend/db/dbgorm/vars.go
Comment (itayd) [reply internal/backend/db/dbgorm/vars.go]: i want to do a one over of all of these in a different pr

File: internal/backend/db/dbgorm/vars.go
Comment (itayd) [reply internal/backend/db/dbgorm/vars.go:36]: now getting all connections in a batch

File: manifest.schema.yaml
Comment (daabr) [inline manifest.schema.yaml:16]: Before merging this, please also prepare a PR that fixes the manifests in https://github.com/autokitteh/samples

File: internal/backend/db/dbgorm/sessions.go
Comment (daabr) [inline internal/backend/db/dbgorm/sessions.go]: More efficient:
```suggestion
	if len(sessions) == int(f.PageSize) && f.PageSize > 0 {
```

File: internal/backend/db/dbfactory/factory.go
Comment (daabr) [inline internal/backend/db/dbfactory/factory.go:27]: I remember that somewhere there was an explicit check of the type value. Do you know where? Is that changed?

File: cmd/ak/cmd/vars/vars.go
Comment (daabr) [inline cmd/ak/cmd/vars/vars.go]: In accordance with #229:

```suggestion
	Use:   "var",
	Short: "Connection variable subcommands: set, get, delete",
```

File: internal/backend/db/dbgorm/sessions.go
Comment (itayd) [reply internal/backend/db/dbgorm/sessions.go]: doesn't belong in this pr

File: cmd/ak/cmd/manifest/apply.go
Comment (daabr) [inline cmd/ak/cmd/manifest/apply.go:13]: 1. This feels like a very specific and hard-to-explain/use flag, but I don't have a good idea to fix that for now
2. Why the focus on just connection vars?
3. "rm" is a very OS-specific term, maybe "del"?

File: cmd/ak/cmd/manifest/plan.go
Comment (daabr) [inline cmd/ak/cmd/manifest/plan.go]: Vars for flags that are shared across commands belong in the parent command, i.e. `manifest.go`

File: cmd/ak/cmd/vars/set.go
Comment (daabr) [inline cmd/ak/cmd/vars/set.go]:
```suggestion
	setCmd.Flags().BoolVarP(&secret, "secret", "s", false, "this value is secret")
```

**Review by daabr — APPROVED** [review]
A few more small comments, but nothing substantial beyond what we discussed. Approving in advance.

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [reply internal/backend/db/dbgorm/vars.go:25]: then we need to ensure that calling function is doing this, which is not the case :)
that's why I suggest to add it here at least

File: internal/backend/db/dbgorm/vars.go
Comment (ig-ra) [reply internal/backend/db/dbgorm/vars.go:21]: @itayd merged the tests into your PR

**Review by ig-ra — APPROVED** [review]

File: internal/backend/db/dbfactory/factory.go
Comment (ig-ra) [reply internal/backend/db/dbfactory/factory.go:27]: I think it's in gormkitteh, but there is a call for `explicit()` which will parse `sqlite` as type from DSN

Comment (itayd) [reply internal/backend/db/dbfactory/factory.go:27]: yeah and he doesn't like it :) wdyt?

Comment (itayd) [reply internal/backend/db/dbfactory/factory.go:27]: let's discuss https://linear.app/autokitteh/issue/ENG-769/gormkitteh-db-type-in-dsn

File: cmd/ak/cmd/manifest/apply.go
Comment (itayd) [reply cmd/ak/cmd/manifest/apply.go:13]: connection vars can be created dynamically by the integration. when applying a manifest - i don't want to delete them, so by default this is not removing any.

i actually think `rm` is nicer than `del`.

File: cmd/ak/cmd/manifest/plan.go
Comment (itayd) [reply cmd/ak/cmd/manifest/plan.go]: moved

File: manifest.schema.yaml
Comment (itayd) [reply manifest.schema.yaml:16]: https://github.com/autokitteh/samples/pull/37

File: internal/backend/db/dbfactory/factory.go
Comment (daabr) [reply internal/backend/db/dbfactory/factory.go:27]: See my comment there

File: cmd/ak/cmd/manifest/apply.go
Comment (daabr) [reply cmd/ak/cmd/manifest/apply.go:13]: This PR is already merged, and I don't have a *strong* opinion, but consider that:
- a Windows user has no idea what "rm" means; "room"? "rem"?
- a Unix user understands what "del" means, even if it's not a Unix command

**Pull request merged**
